lib/Doctrine/ORM/Proxy/Factory/StaticProxyFactory.php

```diff
@@ -186,6 +207,44 @@ private function transientFieldsFqns(ClassMetadata $metadata) : array
         return $transientFieldsFqns;
     }
 
+    private function collectionFieldsFqns(ClassMetadata $metadata) : array
+    {
+        $transientFieldsFqns = [];
+
+        foreach ($metadata->getDeclaredPropertiesIterator() as $name => $property) {
+            if (! $property instanceof ToManyAssociationMetadata) {
+                continue;
+            }
+
+            $transientFieldsFqns[] = $this->propertyFqcn(
+                $property
+                    ->getDeclaringClass()
+                    ->getReflectionClass()
+                    ->getProperty($name) // @TODO possible NPR. This should never be null, why is it allowed to be?
+            );
+        }
+
+        return $transientFieldsFqns;
+    }
+
+    /**
+     * @return ToManyAssociationMetadata[]
+     */
+    private function toManyFields(ClassMetadata $metadata) : array
+    {
+        $toMany = [];
+
+        foreach ($metadata->getDeclaredPropertiesIterator() as $property) {
```

Comment:
Note: not using `array_map()` because this is an iterator

Reply:
There're several pro/cons on Iterator for properties. Keep in mind that `getDeclaredProperties` is different from `getProperties` as the latter also covers hierarchy.

I have yet to decide if we'll make it an iterator forever or no.

Reply:
Ideally, I'd like to have pre-filtered iterators for things like identifiers, association types, etc 😁